javascript/javascript1/week4/index.js

```diff
@@ -0,0 +1,73 @@
+//Voice assistant
+const user = { todo: [] };
+function getReply(command) {
+    let todoItem;
+
+    if (command.startsWith('Hello my name')) {
+        const splitElements = command.split(' ');
+        const indexOfName = splitElements.length - 1;
+        user['myName'] = splitElements.slice(indexOfName);
+        return `Nice to meet you ${user.myName}`;
+    }
+    else if ('What is my name' == command) {
+        return `Your name is ${user['myName']}`;
+    }
+    else if (command.startsWith("Add") && command.endsWith("to my todo")) {
+        todoItem = command.split(' ');
+        const sliceElements = todoItem.slice(1, todoItem.length - 3).join(' ');
+        user.todo.push(sliceElements);
+        return `${sliceElements} added to your todo`;
+    }
+    else if (command.startsWith("Remove") && command.endsWith("for my todo")) {
+        todoItem = command.split(' ');
+        const sliceElements = (todoItem.slice(1, todoItem.length - 3)).join(' ');
+        const index = user.todo.indexOf(sliceElements);
+        if (index > -1) {
+            const itemToDelete = user.todo.splice(index, 1);
+            return `removed ${itemToDelete} from your todo`;
+        }
+    }
+    else if ('What is on my todo?' == command) {
+        const numberOfItemsOnTodo = user.todo.length;
+        return `your to have ${numberOfItemsOnTodo} todos - ${user.todo}`;
+    }
+    else if ('What day is it today?' == command) {
+        const monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+        const dateObj = new Date();
+        const month = monthNames[dateObj.getMonth()];
+        const year = dateObj.getFullYear();
+        const date = dateObj.getDate();
+        return `${date} ${month} ${year}`;
+    }
+    else if (command.startsWith("what is")) {
+        let regExp = /\d [\+\/\*\-]/g;
```

Comment:
If i pass someting like "What is 5 + fewfdsdfdsfds" your function would take it as a valid expression.
Maybe you can try something like this ?
```suggestion
        let regExp = /\d+\s(\+|\/|\*|\-|:)\s\d+/g;
```

Reply:
Thanks Florin. changed the regular expression as you mentioned